malloc/malloc.c

```diff
@@ -29,7 +29,7 @@ char *string_dup(char *src)
     };
     // duplicate now in correct position should have null char
     *copy = '\0';
-    return copy;
+    return copy - length;
```

Comment:
As I told you before, while this may work, it is kind of hacky so definitely not best practice...